### PR TITLE
Fixes #20881 - Clean legacy api from discovery

### PR DESCRIPTION
**Pull request by rabajaj0509**

### CODE REVIEW

File: test/functional/discovered_hosts_controller_test.rb [hunk 1]
```diff
@@ -180,7 +180,7 @@ def test_reboot_success_legacy
     host = discover_host_from_facts(facts)
     Host::Discovered::any_instance.stubs(:proxied?).returns(false)
     Host::Discovered::any_instance.stubs(:proxy_url).returns("http://1.2.3.4:8443")
-    ::ForemanDiscovery::NodeAPI::PowerLegacyDirectService.any_instance.expects(:reboot).returns(true)
+    ::ForemanDiscovery::NodeAPI::PowerService.any_instance.expects(:reboot).returns(true)
     ActiveSupport::Deprecation.silence do
       post "reboot", { :id => host.id }, set_session_user
     end
```
Comment (lzap) [inline test/functional/discovered_hosts_controller_test.rb]: There is also test_reboot_success_legacy test, can you remove that? It is useless I think. Other that that we can merge!

**Review by lzap — CHANGES_REQUESTED** [review]
Looks good let's merge then!

File: app/services/foreman_discovery/node_api/power.rb [hunk 2]
```diff
@@ -5,10 +5,6 @@ def service(data)
         ForemanDiscovery::NodeAPI::PowerService.new(data)
       end
 
-      def legacy_direct_service(data)
-        ForemanDiscovery::NodeAPI::PowerLegacyDirectService.new(data)
-      end
-
       def legacy_proxied_service(data)
         ForemanDiscovery::NodeAPI::PowerLegacyProxiedService.new(data)
       end
```
Comment (lzap) [inline app/services/foreman_discovery/node_api/power.rb]: The legacy_proxied_service can go away too, it should be unused.

File: app/services/foreman_discovery/node_api/power_legacy_direct_service.rb [hunk 3]
```diff
@@ -1,15 +0,0 @@
...
-      raise ArgumentError, "Legacy direct service only supports http scheme" if scheme != 'http'
-      raise ArgumentError, "Legacy direct service only supports port 8443" if port != 8443
-      Foreman::Deprecation.deprecation_warning("1.11", "Discovery legacy API will be removed, use the new Power API")
-    end
-
-    def reboot
-      ::ProxyAPI::BMC.new(:url => url).power :action => "cycle"
-    end
-  end
-end
```
Comment (lzap) [inline app/services/foreman_discovery/node_api/power_legacy_direct_service.rb:15]: Also remove the legacy_proxied.service.rb file please.

Comment (rabajaj0509) [reply app/services/foreman_discovery/node_api/power_legacy_direct_service.rb:15]: @lzap  good catch, thanks :)

**Comment by lzap** [comment]
Ok we are good to go once it is all green.

**Comment by lzap** [comment]
I retriggered failed develop: http://ci.theforeman.org/job/test_plugin_foreman_discovery_develop/324/

**Comment by lzap** [comment]
Foreman core fixed so [test]

**Comment by lzap** [comment]
We can hopefully [test] now.

**Comment by lzap** [comment]
One failure in core is unrelated. Thanks.

**Pull request merged**